### PR TITLE
refactor: reduce number of Cargo.toml files

### PR DESCRIPTION
**Pull request by paulyoong**
Reducing the number of crates makes it easier to manage the versions of
external crates.

The mbus-api, store and types crates have been moved into a single
"control-plane-common" crate.

The deployer directory has also been moved outside of the control-plane
directory.

### CODE REVIEW

**Comment by paulyoong** [comment]
bors try

File: control-plane/agents/Cargo.toml [hunk 1]
```diff
@@ -37,8 +36,7 @@ tracing-futures = "0.2.4"
 rpc = "0.1.0"
 http = "0.2.3"
 paste = "1.0.4"
-store = { path = "../store" }
-types = { path = "../types"}
+control-plane-common = { path = "../../common" }
```
Comment (tiagolobocastro) [inline control-plane/agents/Cargo.toml:39]: should it just be called common?

**Review by tiagolobocastro — APPROVED** [review]

File: control-plane/agents/Cargo.toml (hunk 1 above)
Comment (paulyoong) [reply control-plane/agents/Cargo.toml]: Done - renamed to common-lib to avoid name clash with agents/common.

**Comment by paulyoong** [comment]
bors merge

**Pull request merged**